tracing/src/main/java/io/airlift/tracing/OpenTelemetryModule.java

```diff
@@ -47,11 +50,12 @@ public OpenTelemetryModule(String serviceName, String serviceVersion)
     public void configure(Binder binder)
     {
         newSetBinder(binder, SpanProcessor.class);
+        configBinder(binder).bindConfig(OpenTelemetryConfig.class);
```

Comment:
The existing `OpenTelemetryConfig` is only bound when tracing is enabled. We shouldn't bind it when tracing is disabled, as that would allow configuration of the exporter endpoint, which wouldn't be used.

Instead, let's rename `OpenTelemetryConfig` to `OpenTelemetryExporterConfig`, then create a new `OpenTelemetryConfig` with the sampling config.